MathlibTest/FieldSimp.lean

```diff
@@ -127,11 +148,64 @@ example {hx : x ≠ 0} : P (x / x) := by test_field_simp
 #guard_msgs in
 example {hx : x ≠ 0} : P (x ^ 3 * x⁻¹) := by test_field_simp
 
--- TODO (new implementation): this should reduce to `P (1 / x ^ 2)`
+-- TODO (new implementation): this should reduce to `P (1 / x ^ 3)`
 /-- error: simp made no progress -/
 #guard_msgs in
 example {hx : x ≠ 0} : P (x / x ^ 4) := by test_field_simp
 
+-- We simplify subtracting the same term, even in constants and with literals.
```

Comment:
I would argue this isn't necessarily in scope for `field_simp` -- our new implementation doesn't do it, right? 😄 (E.g. it normalizes `x * (1 - 1)` to `x - x`, not `0`.)

Are you including these tests to document the kind of extra `simp`-ing that `field_simp` currently does and that will be lost?

Reply:
Our implementation could decide to
- unify `x * y` and `y * x` (for example),
- and then decide to cancel `x - x` terms, right?

(It probably doesn't do so, one could argue whether it even should.) In any case, documenting this seems useful.

Reply:
Our implementation normalizes `x * y - y * x` to `x * y * (1 - 1)`, it just doesn't normalize the numeric subtraction to a zero.  Anyway, that decision will get discussed down the line!